kmcd-prelude.md

```diff
@@ -275,5 +276,269 @@ module KMCD-GEN
 
     rule <k> GSS | GSS' => #if R modInt 2 ==K 0 #then GSS #else GSS' #fi ... </k>
          <random> R => randInt(R) </random>
+
+    syntax GenStep ::= GenTimeStep
+    syntax GenTimeStep ::= "GenTimeStep"
+ // ------------------------------------
+    rule <k> GenTimeStep => LogGen ( TimeStep ((I modInt 2) +Int 1) ) ... </k>
+         <random> I => randInt(I) </random>
+
+    syntax GenStep ::= GenVatStep
+    syntax GenVatStep ::= "GenVatFrob"
+                        | "GenVatFrob" CDPID
+                        | "GenVatFrob" CDPID Wad [prefer]
+                        | "GenVatMove"
+                        | "GenVatMove" Address
+                        | "GenVatMove" Address Address
+                        | "GenVatHope"
+                        | "GenVatHope" Address
+                        | "GenVatHope" Address Address
+ // --------------------------------------------------
+    rule <k> GenVatFrob => GenVatFrob chooseCDPID(I, keys_list(VAT_URNS)) ... </k>
+         <random> I => randInt(I) </random>
+         <vat-urns> VAT_URNS </vat-urns>
+      requires size(VAT_URNS) >Int 0
+
+    rule <k> GenVatFrob CDPID => GenVatFrob CDPID ((2 *Rat randRatBounded(I, VAT_GEM)) -Rat VAT_GEM) ... </k>
```

Comment:
I think the expression ((2 *Rat randRatBounded(I, VAT_GEM)) -Rat VAT_GEM) means the value will be in the half-open range [- VAT_GEM, VAT_GEM), which I suppose means that the generator will never consider the case when a user allocates all his gem for dai.